### DISCUSSION TITLE
Telegram WebApp tooling

### DISCUSSION THREAD

**Discussion by v-anyukov** [visciang/telegram#168] [General]
Would someone find useful to have tooling around Telegram WebApp (e.g. validating initData) as part of this package?

**Comment by visciang** [comment]
Hi @v-anyukov,
I've been explicitly kept the library scope small, with focus on Bot API and Bot behaviors.
The API do not expose a typed/structured binding against the BOT HTTP requests but it's fully dynamic and usable against every new bot API version without upgrades (at the price of shifting the request correctness to the client).
So adding support and binding even a small part of the api surface will be in contrast to the library philosophy and also incomplete.

**Reply by v-anyukov** [reply]
Structured bindings is not exactly what I mean. I had in mind something like helper for validating data from mini app as described here https://core.telegram.org/bots/webapps#validating-data-received-via-the-mini-app, may be something else in future. It is too small for dedicated package IMHO and TWA development almost always assumes interaction with the bot - thats how I ended up here with this question.

**Reply by visciang** [reply]
OK, got it!
If it's something "simple" like the utilities in `Telegram.Bot.Utils`, it can definitely fit in.
